lib/node/nodes/kmeans.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var Node = require('../node');
+var dot = require('dot');
+
+var TYPE = 'kmeans';
+
+var PARAMS = {
+    source : Node.PARAM.NODE(Node.GEOMETRY.POINT),
+    clusters : Node.PARAM.NUMBER
+};
+
+var KMeans  = Node.create(TYPE, PARAMS, {cache: false});
```

Comment:
Just pushed a commit setting the caches to True. What should be the strategy for analysis like this that need info from the entire table? I totally agree with your concerns but there are a lot of really powerful analysis methods that are going to need full knowledge of the table, would be good to have a brainstorm at some point about how we might be able to write them in a way thats more efficient and can be made parallelizable